File: CDCAT_ONLINE/AlphaPermute.m
function alpha = AlphaPermute(Q)

% Q is the Q matrix
% alpha is the all possible alpha vectors regarding this number of
% dimension

dim = size(Q,2);
alpha = zeros(1,dim);

for j = 1:dim 
    permute = combnk(1:dim,j);
    alpha0  =zeros(size(permute,1),dim);
    for l = 1:size(alpha0,1)
        alpha0(l,permute(l,:))=1;
    end
    alpha = [alpha;alpha0];
end

File: Sequetial Search Q validation/functions/AlphaPermute.R
###############################################################################
# AlphaPermute:                                                               #
#                                                                             #
# Given the number of attributes, generate all the possible ability patterns. #
#                                                                             #
# Input:                                                                      #
# (1) dim: number of attributes                                               #
#                                                                             #
# Output:                                                                     #
# (1) alpha: a matrix of all the possible ability patterns with rows are      #
#            different patterns and columns are attributes.                   #
###############################################################################

AlphaPermute <- function(dim) {
  
  alpha = rep(0,dim)
  for (j in 1:dim) {
    permute <- t(as.matrix(combn(dim,j)))
    alpha0  <- matrix(0,nrow=nrow(permute),ncol=dim)
    for (l in 1:nrow(alpha0)) {
      alpha0[l,permute[l,]]=1
    }   
    alpha <- rbind(alpha,alpha0)
  }
  
  return(alpha)
}